src/ptracer.rs

```diff
@@ -189,6 +299,32 @@ impl Tracee {
         }
     }
 
+    #[cfg(all(target_os = "android", target_arch = "aarch64"))]
+    pub fn set_debug_register(&self, regtype: DebugRegisterType, index: usize, data: user_hwdebug_state_reg) -> Result<()> {
```

Comment:
As above, I suggest we just have this method write the full debug register struct.

Then we can:
1. Fully reuse `debug_registers()` within this method.
2. Enable callers to efficiently read and write many debug registers at once, with a single syscall sequence.
3. Leave the door open for future Rust wrapper structs that would provide nicer APIs on top of these lower-level methods.

Reply:
Done, except for 1.